draft-ietf-capport-architecture.xml

```diff
@@ -737,6 +737,19 @@ o . . . . . . . . . . . . . . . . . . .| |. . . . . . . . . . . o
      </list>
     </t>
     </section>
+    <section title="Handling of changed Portal URI">
+    <t>This section describes the possibility of changing Portal URI and how to handle
+      this. A different Portal URI could be returned in the following cases:</t>
+    <t>
+      <list style="symbols">
+        <t>If DHCP is used, a lease renewal/rebind may return a different Portal URI.</t>
+        <t>If RA is used, a new Portal URI may be specified in a new RA message received
+          by end user equipment.</t>
+      </list>
+    </t>
+    <t>Whenever a new Portal URI is received by end user equipment, it should discard
+      the old URI and use the new one for future requests to the API.</t>
```

Comment:
Is the URI changing a signal to reauth with the captive portal? I.e. should the UE consider it as a signal to check whether the portal is still authenticated if it is a RUI to the Captive Portal API ?

The fact that there was a change could imply that the UE has associated with a different portal entirely. Should the UE preemptively check whether its session is still valid for good measure?

Reply:
in 7710bis, we have the following:

>   A device may learn about Captive Portal API URIs through more than
>    one of (or indeed all of) the above options.  It is a network
>    configuration error if the learned URIs are not all identical.
> 
>    However, if the URIs learned are not in fact all identical the
>    captive device MUST prioritize URIs learned from network provisioning
>    or configuration mechanisms before all other URIs.  Specifically,
>    URIs learned via any of the options in Section 2 should take
>    precedence over any URI learned via a mechanism like the one
>    described in Section 3.
> 
>    If the URIs learned via more than one option described in Section 2
>    are not all identical, this condition should be logged for the device
>    owner or administrator.  URI precedence in this situation is not
>    specified by this document.

How about we say that the captive portal API URI can not change, if it changes, it's a configuration error (inline with the above quoted paragraphs). In the case of a different API URI is returned in a dhcp renewal or RA, the end user device should ignore it and use the old one instead? (as long as the EU device is still connected to the same network, the first API URI will be used)

Reply:
I can see cases where the captive portal provider may want to change the URI, but there isn't really any guarantee that clients currently connected will be able to detect the change in the URI, so I think it is reasonable that in such cases the provider should maintain both the old and the new one for continuity. So, this is probably a reasonable restriction.

Reply:
On the other hand, I'm not sure what the restriction really provides... Is it just to side-step the question of whether we should reauthenticate? A different sort of signal may be more appropriate in that case, in which case we don't need to avoid the question at all. I think I prefer the text as it is now: just use the latest version. It's simpler.